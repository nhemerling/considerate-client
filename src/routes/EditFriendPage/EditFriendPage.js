import React, { Component } from 'react';
import EditFriendForm from '../../components/EditFriendForm/EditFriendForm';
import FriendApiService from '../../services/friend-api-service';
import ConsiderateContext from '../../context/ConsiderateContext';

export default class EditFriendPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  static contextType = ConsiderateContext;

  handleEditFriendSuccess = (friend) => {
    FriendApiService.getFriends()
      .then((res) => {
        this.context.setFriendList(res);
        const { history } = this.props;
        history.push('/home');
      })
      .catch(this.context.setError);
  };

  render() {
    return (
      <section className="EditFriendPage">
        <h2>Edit Friend</h2>
        <EditFriendForm
          onEditFriendSuccess={this.handleEditFriendSuccess}
          friendId={this.props.match.params.friendId}
        />
      </section>
    );
  }
}
