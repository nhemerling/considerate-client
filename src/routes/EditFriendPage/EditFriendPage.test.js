import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditFriendPage from './EditFriendPage';

describe('EditFriendPage component', () => {
  // simulating the props the component actually has
  const matchProp = {
    params: {
      friendId: 0,
    },
  };

  it('renders a EditFriendPage component', () => {
    const wrapper = shallow(<EditFriendPage match={matchProp} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
