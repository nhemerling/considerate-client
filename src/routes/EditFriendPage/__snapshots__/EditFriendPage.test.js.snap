// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFriendPage component renders a EditFriendPage component 1`] = `
<section
  className="EditFriendPage"
>
  <h2>
    Edit Friend
  </h2>
  <EditFriendForm
    friendId={0}
    onEditFriendSuccess={[Function]}
  />
</section>
`;
